internal-services/catalog/publish-index-image-task.yaml

```diff
@@ -49,14 +49,46 @@ spec:
         quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
       script: |
         #!/usr/bin/env bash
+        set -euo pipefail
+
         PATH=/bin:/usr/bin:/usr/local/bin
         export PATH
 
-        # do not authenticate if the source is redhat's "registry-proxy" which is unauthenticated.
+        AUTH_PARAM=()
+        # Do not authenticate if the source is Red Hat's "registry-proxy" which is unauthenticated.
         if [[ ! "$(params.sourceIndex)" =~ ^registry-proxy(\-stage)?.engineering.redhat.com ]]; then
             AUTH_PARAM=("--src-creds" "${SOURCE_INDEX_CREDENTIAL}")
         fi
+        TARGET_AUTH="--dest-creds ${TARGET_INDEX_CREDENTIAL}"
+
+        echo "Getting source image digest: $(params.sourceIndex)"
+        SOURCE_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${AUTH_PARAM[@]}" \
+            "docker://$(params.sourceIndex)" \
+            --format '{{.Digest}}')
+
+        echo "Getting target image digest: $(params.targetIndex)"
+        if TARGET_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${TARGET_AUTH}" \
+            "docker://$(params.targetIndex)" \
+            --format '{{.Digest}}' 2>/dev/null); then
+            echo "Target image exists."
+            if [ "$SOURCE_DIGEST" == "$TARGET_DIGEST" ]; then
+                echo "Image already exists in target registry with the same digest." \
+                    "Skipping copy." | tee "$(results.requestMessage.path)"
+                exit 0
+            else
+                echo "Image exists in target registry but digests do not match." \
+                    "Proceeding to copy the image." | tee "$(results.requestMessage.path)"
```

Comment:
The results file will get overridden on line 69/101, so there's no use writing anything to it here. If you wanted to have both, you could use `tee -a` at the end.

Reply:
Yeah I don't think this line should be put in the results. The one from line 69/101 is better